server.go

```diff
@@ -1,66 +0,0 @@
-package main
-
-import (
-	"encoding/json"
-	"fmt"
-	"io/ioutil"
-	"net/http"
-	"log"
-)
-
-// *** Structs *** 
-
-type User struct {
-	UserID int 
-	Emails []string 
-}
-
-type Payload struct {
-	UserData User 
-}
-
-// **************
-
-// func serveRest(w http.ResponseWriter, r *http.Request) {
-// 	response, err := getJsonResponse()
-// 	if err != nil {
-// 		panic(err)
-// 	}
-
-// 	fmt.Fprintf(w, string(response))
-// }
-
-func retrieve(w http.ResponseWriter, r *http.Request) {
-	//url := "http://localhost:8081"
-	// res, err := http.Get(url)
-	// if err != nil {
-	// 	panic(err)
-	// }
-	// defer res.Body.Close()
-
-	body, err := ioutil.ReadAll(r.Body)
-	
-	if err != nil {
-		panic(err)
-	}
-
-	var p Payload
-
-	err = json.Unmarshal(body, &p)
-	if err != nil {
-		panic(err)
```

Comment:
When I tried to hit the endpoint from the browser, this `panic(err)` line got executed and the browser said
```
This page isn’t working

localhost didn’t send any data.
ERR_EMPTY_RESPONSE
```

Fortunately go's `http.ListenAndServe` caught the panic and your server recovered. But a nicer way to handle the error is to write a response code back to the client e.g.

```
if err != nil {
    w.WriteHeader(http.StatusBadRequest)
    w.Write([]byte("Unable to parse json body"))
    return
}

w.WriteHeader(http.StatusAccepted)
w.Write([]byte("success"))

```